docs/extensions/redirects.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: MIT
+
+import json
+import os
+from typing import Dict
+
+from _types import SphinxExtensionMeta
+from sphinx.application import Sphinx
+
+
+def main(app: Sphinx, exception: Exception) -> None:
+    # mapping of html node id (i.e., thing after "#" in URLs) to the same but
+    # prefixed with the right doc name
+    # e.g, api.html#disnake.Thread => api/channels.html#disnake.Thread
```

Comment:
this is honestly quite amazing, good job (nothing to change in this comment, just wanted to say this is awesome)